### PR TITLE
Support mlock and custom tmpdir for containerized plugins

### PR DESCRIPTION
**Pull request by tomhjp**
When running containerised plugins within the context of our default systemd unit file (from the rpm repositories), I found a couple of issues I hadn't stumbled on previously:

* Plugins need a capability to mlock if Vault is configured with mlock - previously they always inherited this from Vault as a sub-process, but now we need to explicitly set it for containers. It boils down to adding the IPC_LOCK capability to the container. Without it, we get an error on plugin startup because the mlock syscall fails. Library PR: https://github.com/hashicorp/go-secure-stdlib/pull/94
* If running with systemd's PrivateTmp setting, the container cannot see Vault's view of `/tmp`, so we need the ability to negotiate a separate directory that both sides can see. We _could_ just use TMPDIR, but then that ruins the whole point of the setting. Instead, we introduce a new environment variable to use a different tmp dir in one very narrow use-case. Library PR: https://github.com/hashicorp/go-plugin/pull/282

### CODE REVIEW

File: sdk/helper/pluginutil/run_config.go [hunk 1]
```diff
@@ -139,7 +143,8 @@ func (rc runConfig) makeConfig(ctx context.Context) (*plugin.ClientConfig, error
 		clientConfig.SkipHostEnv = true
 		clientConfig.RunnerFunc = containerCfg.NewContainerRunner
 		clientConfig.UnixSocketConfig = &plugin.UnixSocketConfig{
-			Group: strconv.Itoa(containerCfg.GroupAdd),
+			Group:   strconv.Itoa(containerCfg.GroupAdd),
+			TempDir: os.Getenv("VAULT_PLUGIN_TMPDIR"),
```
Comment (tomhjp) [inline sdk/helper/pluginutil/run_config.go:147]: Maybe this should be settable with both an env var and a config file option?

**Comment by tomhjp** [comment]
Note: I'm planning to merge the library PRs and tag them once all 3 have been reviewed, so once this PR is stable there will be a couple of go.mod changes to use tagged versions of the deps.

File: sdk/helper/pluginutil/run_config.go (hunk 1 above)
Comment (swenson) [reply sdk/helper/pluginutil/run_config.go:147]: It would be nice, but not strictly necessary I think.

**Review by swenson — APPROVED** [review]
LGTM

File: sdk/helper/pluginutil/run_config.go (hunk 1 above)
Comment (tomhjp) [reply sdk/helper/pluginutil/run_config.go:147]: 👍 I think I'll aim to add a config option for 1.15.1 given where we are in the release cycle.

**Comment by tomhjp** [comment]
Thanks!

**Pull request merged**